packages/teuchos/parameterlist/test/ParameterList/ParameterList_UnitTests.cpp

```diff
@@ -70,6 +71,74 @@ class DummyValidator : public Teuchos::ParameterEntryValidator
 };
 
 
+class SimpleModifier : public Teuchos::ParameterListModifier
+{
+public:
+
+  SimpleModifier() : Teuchos::ParameterListModifier("Simple Modifier"){}
+
+  void modify(Teuchos::ParameterList &pl, Teuchos::ParameterList &valid_pl) const
+  {
+    expandSublistsUsingBaseName("SubA", pl, valid_pl);
+  }
+
+  void reconcile(Teuchos::ParameterList &pl) const
+  {
+    // If A and B are less than 0.0 then throw an error
+    if (pl.get<double>("A") < 0.0 && pl.get<double>("B") < 0.0){
+      throw std::logic_error("Parameters A and B can't both be less than 0.0");
```

Comment:
Please use `TEUCHOS_TEST_FOR_EXCEPTION()` instead of raw throws.

Reply:
Yes, this is just test code but using `TEUCHOS_TEST_FOR_EXCEPTION()` will make debugging easier on some systems.

Reply:
Fixed in next push